bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/ReadHandle.java

```diff
@@ -111,4 +112,32 @@
      */
     long getLength();
 
+    /**
+     * The type contains LAC and a LedgerEntry want to read.
+     * It is used for readLastAddConfirmedAndEntry.
+     */
+    @Data
+    class LastConfirmedAndEntry {
```

Comment:
I think it is better that this class is not an inner class but a top level class

Reply:
move it out of ReadHandle.java, make it an interface

```
interface LastConfirmedAndEntry extends AutoCloseable {

     long getLastAddConfirmed();

     boolean hasEntry();

     LedgerEntry getEntry();

}
```

create a class LastConfirmedAndEntryImpl to implement the interface.

Reply:
Thanks @eolivelli @sijie , will change it.